cardano-api/test/Test/Cardano/Api/MetaData.hs

```diff
@@ -6,34 +6,37 @@ module Test.Cardano.Api.MetaData
 
 import           Cardano.Prelude hiding (MetaData)
 
-import           Cardano.Api.Typed
 import           Cardano.Api.MetaData
+import           Cardano.Api.Typed
 
+import qualified Data.ByteString.Base16 as Base16
 import qualified Data.ByteString.Char8 as BS
-import qualified Data.Map.Strict as Map
 import qualified Data.List as List
+import qualified Data.Map.Strict as Map
 import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
 
 import           Hedgehog (Gen, Property, discover)
 import qualified Hedgehog
 import qualified Hedgehog.Gen as Gen
-import           Hedgehog.Range (Range)
+import qualified Hedgehog.Internal.Gen as Gen
 import qualified Hedgehog.Range as Range
 
 -- This test is fragile to changes in the way Metadata is converted to/from JSON.
 prop_round_trip_json_metadatum :: Property
 prop_round_trip_json_metadatum = do
   Hedgehog.withTests 1000 . Hedgehog.property $ do
-    md <- Hedgehog.forAll genMetaData
-    Hedgehog.tripping md jsonFromMetadata jsonToMetadata
+    json <- jsonFromMetadata <$> Hedgehog.forAll genMetaData
+    Hedgehog.tripping json identity (fmap jsonFromMetadata . jsonToMetadata)
```

Comment:
What? Why is `identity` being used there? That defeats the prurpose of the `tripping` function's ability to pretty print failed test cases.

Reply:
Fixed - we need to convert the generated metadata to JSON before round-trip testing to "wash out" ambiguous metadatums. For example, `MD.S "0x"` becomes `MD.B ""` after a round-trip, causing the property to fail. I don't really like this though.